Console/UI/Pages/EventsLog.xaml.cs

```diff
@@ -126,94 +140,114 @@ public EventsLog()
+                securityLog.EndInit();
+
             }
             catch (Exception exc)
             {
                 LogHelper.Error("Unable to connect to the event log", exc);
                 throw;
             }
 
-            Task.Run(() => InitEventLog(value => ScanProgress = value));
+        }
+        private void PauseHandlingSecurityLogEvents()
+        {
+            securityLog.EnableRaisingEvents = false;
+        }
+        private void StopHandlingSecurityLogEvents()
+        {
+            PauseHandlingSecurityLogEvents();
+            securityLog?.Dispose();
         }
 
         private void SecurityLog_EntryWritten(object sender, EntryWrittenEventArgs e)
         {
+            //LogHelper.Debug($"EntryWritten event ...");
             if (!FirewallHelper.IsEventAccepted(e.Entry))
             {
                 return;
             }
 
-            LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
-            if (entry != null)
+            lock (EntriesLock)
             {
-                Dispatcher.Invoke(() =>
+                if (e.Entry != null && e.Entry.Index > SecurityLogEntryIndex_last)
                 {
-                    if (!TargetCollectionNewEntries.Contains(entry, EntryComparer.Instance))
+                    LogEntryViewModel entry = EntryViewFromLogEntry(e.Entry);
+                    if (entry != null)
                     {
-                        TargetCollectionNewEntries.Add(entry);
+                        Dispatcher.Invoke(() =>
+                        {
+                            Dispatcher.Invoke(() => { LogEntries.Add(entry); ScanProgress = LogEntries.Count; });
+                        });
                     }
-                });
+                }
+                if (ScanProgressMax < LogEntries.Count)
+                {
+                    // overflow handling
+                    LogEntries.RemoveAt(0);
+                }
+                SecurityLogEntryIndex_last = securityLog.Entries.Count - 1;
             }
         }
 
-        private static bool TCPFilter(object le) => ((LogEntryViewModel)le).Protocol == "TCP";
-
-        private void InitEventLog(Action<int> progressCallback)
+        /*
+         * Initializes the LogEntries list to show the most recent events
+         */
+        private void InitEventLog(int lastIndex, Action<int> progressCallback)
         {
+            int i = 0;
             try
             {
-                Dispatcher.Invoke(() => LogEntries.Clear());
-
-                ScanProgressMax = securityLog.Entries.Count;
-
-                // Have to use a Enumerable.Range to enumerate the entries starting from the last one, since calling Reverse on the initial collection will first enumerate all entries
-                var entries = Enumerable.Range(0, ScanProgressMax)
-                                        .Select(i => { progressCallback?.Invoke(i + 1); return securityLog.Entries[ScanProgressMax - i - 1]; })

```

Comment:
btw: one of the problems IndexOutOfBounds was here:
>.Select(i => { return securityLog.Entries[ScanProgressMax - i - 1]; })

Should have been:

>.Select(i => { return securityLog.Entries[ScanProgressMax - (i+1)]; })

Reply:
It's the exact same thing ;-)